apps/web/src/containers/WalletContainer.ts

```diff
@@ -92,80 +93,83 @@ function useWallet(initialState: State = defaultState): DepositContainerState {
     state: { currentAccount, signer },
   } = Web3Container.useContainer()
   const [error, setError] = useState<Error | undefined>()
-  const [loading, setLoading] = useState<boolean>(false)
-
-  const { data, loading: reserveLoading } = useQuery<
-    ReservesGraph,
-    UserReserveVariables
-  >(GET_RESERVE_DATA, {
-    variables: {
-      user: currentAccount?.toLowerCase(),
+  const [loading, setLoading] = useState<boolean>(true)
+
+  const { data } = useQuery<ReservesGraph, UserReserveVariables>(
+    GET_RESERVE_DATA,
+    {
+      variables: {
+        user: currentAccount?.toLowerCase(),
+      },
     },
-  })
+  )
 
+  const getReserveBalance = async (name: TokenList) => {
+    try {
+      const tokenContract: TokenContract = Tokens.getTokenContract(signer, name)
+
+      const balance = await Tokens.getTokenBalance(
+        tokenContract,
+        currentAccount,
+      )
+      setError(undefined)
+      return balance
+    } catch (error) {
+      setError(error)
+    }
+  }
   const setReserveData = useCallback(
-    async (reserve: ReservesData) => {
-      setLoading(true)
-      try {
-        const tokenContract: TokenContract = Tokens.getTokenContract(
-          signer,
-          reserve.name,
-        )
-
-        const balance = await Tokens.getTokenBalance(
-          tokenContract,
-          currentAccount,
-        )
-
-        dispatch({
-          type: 'setReserveData',
-          value: {
+    async (reserves: any) => {
+      const reserveData = await Promise.all(
```

Comment:
we should still have try, catch around, because we are not handling now if one of the primises is rejected.